src/textual/widgets/_collapsible.py

```diff
@@ -99,6 +99,23 @@ class Collapsible(Widget):
     }
     """
 
+    class Toggled(Message):
```

Comment:
I'd observe that leaning into a single message rather than one for expanded and one for collapsed makes it just a touch harder to write more declarative code with the `@on` decorator; but I'm not so bothered about it as to heavily advocate for it.

Reply:
How about if `Collapsed` and `Expanded` inherit from `Toggled`, so that we can actually get the best of both worlds?

Reply:
Exactly what I was thinking. :-)

Reply:
Do you mind taking another look just to make sure my change ba9c00e was in the right direction?

Reply:
Done; double approved.